dds_cli/user.py

```diff
@@ -312,40 +309,30 @@ def token_report(self, token_metadata=None):
         LOG.info(f"[{markup_color}]{sign}  {message} {sign} [/{markup_color}]")
         LOG.info(f"[{markup_color}]Token age: {readable_timedelta(age)}[/{markup_color}]")
 
-        if lifetime.total_seconds() < 0:
+        if age > dds_cli.TOKEN_MAX_AGE:
+            LOG.info(f"[{markup_color}]Token expired: {expiration_time}[/{markup_color}]")
+        else:
+            lifetime = age - dds_cli.TOKEN_MAX_AGE
             LOG.info(
                 f"[{markup_color}]Token expires: {expiration_time} (in {readable_timedelta(lifetime)})[/{markup_color}]"
             )
-        else:
-            LOG.info(f"[{markup_color}]Token expired: {expiration_time}[/{markup_color}]")
-
-        if consignee:
-            LOG.info(f"[{markup_color}]Token issued to: {consignee}[/{markup_color}]")
 
     # Private methods ############################################################ Private methods #
-    def __token_dates(self, token_metadata):
-        """Returns definitive or estimated values for the issue date, token's age, lifetime and expiration time in UTC."""
-
-        local_tz = tzlocal.get_localzone()
-        utc_tz = pytz.timezone("UTC")
-
-        # Try to use the Issued At Claim (iat), otherwise fall back to modification_time of the file
-        if token_metadata and token_metadata.get("iat", None):
-            issued_at = datetime.datetime.fromtimestamp(token_metadata.get("iat"), tz=utc_tz)
-        else:
-            issued_at = datetime.datetime.fromtimestamp(
-                os.path.getmtime(self.token_file), tz=local_tz
-            )
-
-        age = datetime.datetime.utcnow().replace(tzinfo=utc_tz) - issued_at
-
-        # Try to use the Expiration Time (exp), otherwise fall back to calculation based on configured lifetime value
-        if token_metadata and token_metadata.get("exp", None):
-            expiration_time = datetime.datetime.fromtimestamp(token_metadata.get("exp"), tz=utc_tz)
+    def __token_dates(self, token):
+        """Returns definitive (based on the token jose header) or estimated values
+        (based on the token file) for the token's age and expiration time in UTC."""
+
+        expiration_time = get_token_expiration_time(token=token)
+
+        # Try to use the Expiration Time (exp) in the token jose header,
+        # otherwise fall back to calculation based on the token file
+        if expiration_time:
+            utc_tz = pytz.timezone("UTC")
+            expiration_time = datetime.datetime.fromtimestamp(expiration_time, tz=utc_tz)
+            age = dds_cli.TOKEN_MAX_AGE - (expiration_time - datetime.datetime.utcnow())
         else:
-            expiration_time = issued_at + dds_cli.TOKEN_MAX_AGE
-
-        # lifetime is returned as a negative value / countdown
-        lifetime = datetime.datetime.utcnow().replace(tzinfo=utc_tz) - expiration_time
+            modification_time = datetime.datetime.fromtimestamp(os.path.getmtime(self.token_file))
```

Comment:
Mind that in the current version I correct for the local timezone, since `os.path.getmtime` returns the time in the local system time. So you need to have a time zone correction here!

I have tried it out and without using `tzlocal.get_localzone()` there was an offset of 1h due to Stockholm time. Should we ever have users located in Japan or the US, that will of course be more significant.